doc/adding_lints.md

```diff
@@ -309,6 +373,19 @@ declare_clippy_lint! { /* ... */ }
 Once your lint is merged, this documentation will show up in the [lint
 list][lint_list].
 
+### Running rustfmt
+
+[Rustfmt](https://github.com/rust-lang/rustfmt) is a tool for formatting Rust code according
+to style guidelines. Your code has to be formatted by `rustfmt` before a PR can be merged.
+
+It can be installed via `rustup`:
+
+```bash
+rustup component add rustfmt
+```
+
+Use `cargo fmt --all` to format the whole codebase.
```

Comment:
I recently realized, that this doesn't format the test files. 

The workaround in travis is to walk the `tests` folder and call `rustfmt` on each test file:
https://github.com/rust-lang/rust-clippy/blob/9f2055637e3a32064517c888f7d69b4fad6d67c5/ci/base-tests.sh#L62-L68

Maybe we should provide a script which does just that.

```bash
cargo fmt --all
for file in `find tests -not -path "tests/ui/doc.rs" -not -path "tests/ui/unused_unit.rs" | grep "\.rs$"` ; do 
    rustfmt ${file}
done
```

Reply:
Yeah, that's something that would be super nice! I created https://github.com/rust-lang/rust-clippy/issues/3835 to further track this.